File: code/jvm/im-repository/src/main/kotlin/im/pagination/PaginationInfo.kt
package im.pagination

/**
 * Represents pagination information.
 *
 * @param total the total number of items
 * @param currentPage the current page number
 * @param totalPages the total number of pages
 * @param nextPage the next page number or null if there is no next page
 * @param prevPage the previous page number or null if there is no previous page
 */
data class PaginationInfo(
    val total: Long,
    val currentPage: Int,
    val totalPages: Int,
    val nextPage: Int?,
    val prevPage: Int?
)

File: code/client/web/src/Domain/pagination/PaginationInfo.ts
/**
 * Interface for pagination information
 *
 * @param total The total number of items.
 * @param currentPage The current page number.
 * @param totalPages The total number of pages.
 * @param nextPage The next page number.
 * @param prevPage The previous page number.
 *
 */
export interface PaginationInfo {
    total: number | null;
    totalPages: number | null;
    current: number;
    next: number | null;
    previous: number | null;
}
